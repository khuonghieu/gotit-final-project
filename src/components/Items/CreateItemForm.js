import { connect } from 'react-redux';
import { Form, Button } from '@gotitinc/design-system';
import React, { useState } from 'react';
import { createItem } from '../../actions/items';

function mapStateToProps(state) {
  return {
    currentCategory: state.categories.currentCategory,
  };
}

const mapDispatchToProps = (dispatch) => ({
  createItem: (categoryId, name, description, price) => dispatch(createItem(categoryId, name, description, price)),
});

export function CreateItemForm({ currentCategory, createItem }) {
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [price, setPrice] = useState(0.0);

  async function handleCreateItem(e) {
    e.preventDefault();
    if (currentCategory && name && description && price) {
      const { success, payload } = await createItem(currentCategory, name, description, price);
      if (!success) {
        console.log(payload);
      }
    } else {
      alert('Fill all the blanks');
    }
  }
  return (
    <div>
      <Form.Label>Item name</Form.Label>
      <Form.Input value={name} onChange={(e) => setName(e.target.value)} type="text" placeholder="Enter item name" name="name" />
      <Form.Label>Item description</Form.Label>
      <Form.Input value={description} onChange={(e) => setDescription(e.target.value)} type="text" placeholder="Enter item description" name="description" />
      <Form.Label>Item price</Form.Label>
      <Form.Input value={price} onChange={(e) => setPrice(e.target.value)} type="number" placeholder="Enter item price" name="price" />
      <Button onClick={handleCreateItem}>Submit</Button>
    </div>
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(CreateItemForm);
