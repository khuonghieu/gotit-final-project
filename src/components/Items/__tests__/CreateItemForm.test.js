import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button, Form } from '@gotitinc/design-system';
import { CreateItemForm } from '../CreateItemForm';

configure({ adapter: new Adapter() });

describe('components/Items/CreateItemForm.js', () => {
  let wrapper;
  let props;
  let input;
  let button;

  const update = () => {
    input = wrapper.find(Form.Input);
    button = wrapper.find(Button);
  };

  beforeEach(() => {
    props = {
      currentCategory: 1,
      createItem: jest.fn(),
      refreshItemList: jest.fn(),
    };
  });

  const setup = () => {
    wrapper = shallow(
      <CreateItemForm {...props} />,
    );
    update();
  };

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });

  // it('should call SignIn API when form is fully filled and clicked submit button', () => {
  //   setup();
  //   input.at(0).simulate('change', { target: { value: 'testname' } });
  //   input.at(1).simulate('change', { target: { value: 'testdescription' } });
  //   input.at(2).simulate('change', { target: { value: 5 } });
  //   button.simulate('click');
  //   expect(props.createItem).toHaveBeenCalled();
  // });

  // it('should refresh item list when done creating item', () => {
  //   setup();
  //   input.at(0).simulate('change', { target: { value: 'testname' } });
  //   input.at(1).simulate('change', { target: { value: 'testdescription' } });
  //   input.at(2).simulate('change', { target: { value: 5 } });
  //   button.simulate('click');
  //   expect(props.refreshItemList).toHaveBeenCalled();
  // });
});
