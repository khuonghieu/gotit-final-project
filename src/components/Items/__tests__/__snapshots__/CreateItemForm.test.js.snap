// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Items/CreateItemForm.js should render correctly 1`] = `ShallowWrapper {}`;
